lib/addons/src/hooks/useWindowSize.ts

```diff
@@ -0,0 +1,36 @@
+import { useState, useEffect, useCallback } from 'react';
+import root from 'window-or-global';
+
+export interface WindowSize {
+  width: number | undefined;
+  height: number | undefined;
+}
+
+// Hook
+export const useWindowSize = () => {
+  // Initialize state with undefined width/height so server and client renders match
+  // Learn more here: https://joshwcomeau.com/react/the-perils-of-rehydration/
```

Comment:
We don't SSR?

Reply:
I don't know if we do or not

Just though I would add this as it would be exposed to the outside world